dom.bs

```diff
@@ -1413,6 +1413,212 @@ can only be used to influence an ongoing one.
+
+<ol>
+ <li><p>If <var>signal</var>'s <a for=AbortSignal>aborted flag</a> is set, then return.
+
+ <li><p><a for=set>Append</a> <var>algorithm</var> to <var>signal</var>'s
+ <a for=AbortSignal>abort algorithms</a>.
+</ol>
+
+<p>To <dfn export for=AbortSignal>remove</dfn> an algorithm <var>algorithm</var> from an
+{{AbortSignal}} <var>signal</var>, <a for=set>remove</a> <var>algorithm</var> from
+<var>signal</var>'s <a for=AbortSignal>abort algorithms</a>.
+
+<p class="note no-backref">The [=AbortSignal/abort algorithms=] enable APIs with complex
+requirements to react in a reasonable way to {{AbortController/abort()}}. For example, a given API's
+[=AbortSignal/aborted flag=] might need to be propagated to a cross-thread environment, such as a
+service worker.
+
+<p>The <dfn attribute for=AbortSignal>aborted</dfn> attribute's getter must return true if
+<a>context object</a>'s [=AbortSignal/aborted flag=] is set, and false otherwise.
+
+<p class=note>Changes to an {{AbortSignal}} object represent the wishes of the corresponding
+{{AbortController}} object, but an API observing the {{AbortSignal}} object can chose to ignore
+them. For instance, if the operation has already completed.
+
+<p>To <dfn export for=AbortSignal>signal abort</dfn>, given a {{AbortSignal}} object
+<var>signal</var>, run these steps:
+
+<ol>
+ <li><p>If <var>signal</var>'s [=AbortSignal/aborted flag=] is set, then return.
+
+ <li><p>Set <var>signal</var>'s [=AbortSignal/aborted flag=].
+
+ <li><p><a for=set>For each</a> <var>algorithm</var> in <var>signal</var>'s
+ [=AbortSignal/abort algorithms=]: run <var>algorithm</var>.
+
+ <li><p><a for=set>Empty</a> <var>signal</var>'s <a for=AbortSignal>abort algorithms</a>.
+
+ <li><p>[=Fire an event=] named <code event for=AbortSignal>abort</code> at <var>signal</var>.
+</ol>
+
+
+<h3 id=abortcontroller-api-integration>Using {{AbortController}} and {{AbortSignal}} objects in
+APIs</h3>
+
+<p>Any web platform API using promises to represent operations that can be aborted must adhere to
+the following:
+
+<ul class=brief>
+ <li>Accept {{AbortSignal}} objects through a <code>signal</code> dictionary member.
+ <li>Convey that the operation got aborted by rejecting the promise with an "{{AbortError}}"
+ {{DOMException}}.
+ <li>Reject immediately if the {{AbortSignal}}'s [=AbortSignal/aborted flag=] is already set,
+ otherwise:
+ <li>Use the [=AbortSignal/abort algorithms=] mechanism to observe changes to the {{AbortSignal}}
+ object and do so in a manner that does not lead to clashes with other observers.
+</ul>
+
+<div class=example id=aborting-ongoing-activities-spec-example>
+ The steps for a promise-returning method <code>doAmazingness(options)</code> could be as follows:
+
+ <ol>
+  <li>Let |p| be [=a new promise=].
+  <li>If |options|' <code>signal</code> member is present, then:
+   <ol>
+    <li>If |options|' <code>signal</code>'s [=AbortSignal/aborted flag=] is set, throw an
+    "{{AbortError}}" {{DOMException}}.
```

Comment:
Is it worth adding a note here to point out this will be turned into a rejection, or is that assumed knowledge?